System/Console/ANSI/Types.hs

```diff
@@ -37,28 +38,26 @@ data ColorIntensity = Dull
                     | Vivid
                     deriving (Eq, Ord, Bounded, Enum, Show, Read, Ix)
 
-
+-- | ANSI allows for a palette of up to 256 colors
```

Comment:
Maybe say "256 8-bit colors", and also why up-to? do you mean that there are duplicate colors? It could be the case for rgb that have the same r,g,b maybe and grayscale?

Reply:
I said 'up to' because I understand that some terminals (RXVT?) offered only an 88-colour palette - although I suspect that 88-colour palettes are ancient history now. The extract from the `XTerm Control Sequences` guide (http://invisible-island.net/xterm/ctlseqs/ctlseqs.html) reads:
~~~~
Xterm maintains a color palette whose entries are identified
by an index beginning with zero.  If 88- or 256-color support
is compiled, the following apply:
o All parameters are decimal integers.
o RGB values range from zero (0) to 255.
o ISO-8613-3 can be interpreted in more than one way; xterm
allows the semicolons in this control to be replaced by
colons (but after the first colon, colons must be used).

These ISO-8613-3 controls are supported:
Pm = 3 8 ; 2 ; Pr; Pg; Pb -> Set foreground color to the
closest match in xterm's palette for the given RGB Pr/Pg/Pb.
Pm = 3 8 ; 5 ; Ps -> Set foreground color to Ps.
Pm = 4 8 ; 2 ; Pr; Pg; Pb -> Set background color to the
closest match in xterm's palette for the given RGB Pr/Pg/Pb.
Pm = 4 8 ; 5 ; Ps -> Set background color to Ps.
~~~~
The xterm 256 colour protocol has a small number of duplicated colours in its palette - black, white and the non-grey vivid system colours all appear twice.